// node-reminders/build/main/lib/reminders.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("./utils");
function getLists() {
    return utils_1.execJXA(utils_1.JXA_SCRIPTS.getLists);
}
exports.getLists = getLists;
function getList(id) {
    return utils_1.execJXA(utils_1.JXA_SCRIPTS.getList, { id });
}
exports.getList = getList;
async function createList(data) {
    // tslint:disable-next-line: no-expression-statement
    await utils_1.execJXA(utils_1.JXA_SCRIPTS.createList, { data });
    const lists = await getLists();
    return lists[lists.length - 1].id;
}
exports.createList = createList;
async function getReminders(listId, props) {
    const reminders = await utils_1.execJXA(utils_1.JXA_SCRIPTS.getReminders, {
        id: listId,
        props,
    });
    return reminders.map(utils_1.withParsedDates);
}
exports.getReminders = getReminders;
async function getReminder(reminderId, props) {
    const reminder = await utils_1.execJXA(utils_1.JXA_SCRIPTS.getReminder, { id: reminderId, props });
    return utils_1.withParsedDates(reminder);
}
exports.getReminder = getReminder;
function updateReminder(id, data) {
    return utils_1.execJXA(utils_1.JXA_SCRIPTS.updateReminder, { id, data });
}
exports.updateReminder = updateReminder;
function deleteReminder(id) {
    return utils_1.execJXA(utils_1.JXA_SCRIPTS.deleteReminder, { id });
}
exports.deleteReminder = deleteReminder;
function createReminder(listId, data) {
    return utils_1.execJXA(utils_1.JXA_SCRIPTS.createReminder, { listId, data });
}
exports.createReminder = createReminder;
