// node-reminders/build/main/lib/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const execa_1 = __importDefault(require("execa"));
const path = __importStar(require("path"));
const JXA_SCRIPTS_DIRNAME = 'jxa';
function jxaPath(name) {
    return path.join(__dirname, '..', JXA_SCRIPTS_DIRNAME, `${name}.jxa`);
}
exports.JXA_SCRIPTS = {
    createList: jxaPath('create-list'),
    createReminder: jxaPath('create-reminder'),
    deleteReminder: jxaPath('delete-reminder'),
    getList: jxaPath('get-list'),
    getLists: jxaPath('get-lists'),
    getReminder: jxaPath('get-reminder'),
    getReminders: jxaPath('get-reminders'),
    updateReminder: jxaPath('update-reminder'),
};
async function execJXA(scriptPath, data) {
    const prefix = process.env.NODE_ENV === 'test' ? 'test' : '';
    const { stderr } = await execa_1.default(prefix + 'osascript', [
        '-l',
        'JavaScript',
        scriptPath,
        JSON.stringify(data),
    ]);
    return JSON.parse(stderr);
}
exports.execJXA = execJXA;
/**
 * Takes an object and return an equivalent object with string dates converted to Date objects
 */
function withParsedDates(data) {
    const dateProps = [
        'completionDate',
        'creationDate',
        'dueDate',
        'modificationDate',
        'remindMeDate',
    ];
    const final = Object.keys(data).reduce((obj, prop) => {
        const isDate = dateProps.indexOf(prop) !== -1;
        const val = data[prop];
        const newObj = Object.assign(Object.assign({}, obj), { [prop]: isDate ? new Date(val) : val });
        return newObj;
    }, {});
    return final;
}
exports.withParsedDates = withParsedDates;
